src/ray/raylet/node_manager.cc

```diff
@@ -494,10 +495,18 @@ void NodeManager::DispatchTasks() {
       continue;
     }
     // We have enough resources for this task. Assign task.
-    // TODO(atumanov): perform the task state/queue transition inside AssignTask.
-    // (See design_docs/task_states.rst for the state transition diagram.)
-    auto dispatched_task = local_queues_.RemoveTask(task.GetTaskSpecification().TaskId());
-    AssignTask(dispatched_task);
+    // (AssignTask() takes a non-const so copy task in a non-const variable.)
```

Comment:
Hmm actually looking at `AssignTask()`, it doesn't seem necessary to have it take in a non-const. It modifies the `Task` but only in the handler for writing the message to the worker, which has to make a copy of the task anyway ([code](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L1297)).

Reply:
Just to make sure. You are proposing to (1) make the argument of AssignTask() `const`, and then (2) make a copy of `task` when passing to the callback of `WriteMessageAsync()`?